types/three/test/unit/src/nodes/core/StructNode.ts

```diff
@@ -0,0 +1,69 @@
+
+import { Fn, vec3, struct, float } from "three/tsl";
+
+
+const Complex = struct({
+    real:'float',
+    imaginary:'float',
+});
+
+const multiply = Fn(([a,b]:[a:ReturnType<typeof Complex>,b:ReturnType<typeof Complex>])=>{
+    const ar=a.get("real");
+    const ai=a.get("imaginary");
+    const br=b.get("real");
+    const bi=b.get("imaginary");
+    return Complex({
+        real: ar.mul(br).sub(ai.mul(bi)),
+        imaginary: ar.mul(bi).add(ai.mul(br))
+    });
+})
```

Comment:
Thanks for the PR! Unfortunately I don't think this is strict enough. I think ideally we would want `a.get("invalid")` to produce a type-error, but it looks like it current allows any key.